handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -494,6 +494,12 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         }
         try {
             wrap(ctx, false);
+        } catch (Exception e) {
+            // Just fail the pending writes in case of an SSLException. If there are no pending writes rethrow.
+            if (pendingUnencryptedWrites.removeAndFailAll(e)) {
```

Comment:
@Scottmitch its because if we have pending writes we should just fail these. Its the same reason why we fail a promise in write(...) and not throw an exception. Makes sense ?